src/middlewared/middlewared/plugins/kubernetes_linux/k8s/watch.py

```diff
@@ -0,0 +1,52 @@
+import contextlib
+
+import asyncio
+import json
+import typing
+
+
+from .client import ClientMixin
+from .core_api import Event, Pod
+
+
+class Watch(ClientMixin):
+
+    def __init__(
+        self, resource: typing.Union[typing.Type[Event], typing.Type[Pod]],
+        resource_uri_args: typing.Optional[dict] = None,
+    ):
+        self.resource: typing.Union[typing.Type[Event], typing.Type[Pod]] = resource
+        self.resource_ui_args: typing.Optional[dict] = resource_uri_args or {}
+        self._stop: bool = False
+
+    async def stop(self) -> None:
+        self._stop = True
+
+    @classmethod
+    def sanitize_data(cls, data: bytes, response_type: str) -> typing.Union[dict, str]:
```

Comment:
You allow binary data that fails to be decoded from utf-8 to be returned as just binary data, is this intentional?

Reply:
It was just an extra fail safe but of course it would break at consumer side then.
I have modified it now and going to raise an exception to fail hard if this happens.